src/processors/knex-processor.ts

```diff
@@ -147,18 +151,18 @@ export default class KnexProcessor<
     }
 
     return await this.getQuery()
-      .where({ id })
+      .where({ [primaryKeyName]: id })
       .select(getColumns(this.resourceClass))
       .first();
   }
 
   async add(op: Operation): Promise<HasId> {
     const { type } = op.ref;
-
-    const ids = await this.getQuery().insert(op.data.attributes, "id");
+    const primaryKeyName = this.resourceClass.schema.primaryKeyName || "id";
```

Comment:
@spersico Seems like we could make the `schema` assume `id` has default so this defaulting can be done only once?